src/deadline_worker_agent/sessions/session.py

```diff
@@ -1140,6 +1159,42 @@ def _action_updated_impl(
 
         if self._output_sync_target_action is not None:
             if OPENJD_ACTION_STATE_TO_DEADLINE_COMPLETED_STATUS.get(action_status.state, None):
+                manifests_list = None
+
+                # Get job attachment details to access input manifests
+                job_attachment_details = self._job_attachment_details
+
+                if job_attachment_details:
+                    # Create a list of ManifestInfo objects with the same length as the input manifests list
+                    manifests_list = []
+
+                    # For each input manifest, find the corresponding output manifest
+                    for input_manifest in job_attachment_details.manifests:
+                        asset_root = input_manifest.root_path
+                        manifest_info = None
+
+                        for output_manifest in self._upload_manifest_list:
+                            if (
+                                output_manifest.source_path is not None
+                                and output_manifest.source_path == asset_root
+                            ):
+                                manifest_info = output_manifest
+                                break
+
+                        # Create a ManifestInfo dictionary with the appropriate values
+                        manifest_info_obj: ManifestInfo = {}
+
+                        if manifest_info and manifest_info.output_manifest_path is not None:
+                            manifest_info_obj["outputManifestPath"] = (
+                                manifest_info.output_manifest_path
+                            )
+                        if manifest_info and manifest_info.output_manifest_hash is not None:
+                            manifest_info_obj["outputManifestHash"] = (
+                                manifest_info.output_manifest_hash
+                            )
```

Comment:
What happens if only one of `outputManifestPath` or `outputManifestHash` are in the `manifest_info` object?